python/desc/imsim/ImageSimulator.py

```diff
@@ -103,6 +110,10 @@ def __init__(self, instcat, psf, numRows=None, config=None, seed=267,
         self._make_gs_interpreters(seed, sensor_list, file_id)
         self.log_level = log_level
         self.logger = get_logger(self.log_level, name='ImageSimulator')
+        self.ckpt_archive_dir = ckpt_archive_dir
+        if (self.ckpt_archive_dir is not None and
+            not os.path.isdir(self.ckpt_archive_dir)):
+            os.makedirs(self.ckpt_archive_dir)
```

Comment:
I'm wondering if this should be in a try: block?

Im thinking that people might put down general shared directories here, and it might turn out they don't exist or aren't available in certain run conditions or systems.

Reply:
I'm not sure what the action should be if it catches an error.  Generally, unless there's a very specific action to take that resolves the error cleanly, I think it's better to let any exceptions go uncaught so that the failure mode is clear at runtime.

Reply:
Ah... well I was thinking of catching it, making a very clear statement about went wrong and then stopping the program.  I thought this might be the sort of thing that was confusing to non-experts if they just saw the stack-trace.

So, basically the same but trying to be a bit more user friendly.  But, I suppose you are right that the standard error will have more information in it for those who can read it.